### PR TITLE
Update fluentd-gcp DaemonSet

### PR DESCRIPTION
**Pull request by tallclair**
- Use a dedicated service account to run the fluentd-gcp DS
- Use the certificates in the prometheus-to-sd image rather than mounting the host certs

This PR lets us create a more targeted PodSecurityPolicy for fluentd. (See https://github.com/kubernetes/kubernetes/pull/52367#discussion_r145433354)

```release-note
- fluentd-gcp runs with a dedicated fluentd-gcp service account
- Stop mounting the host certificates into fluentd's prometheus-to-sd container
```

### CODE REVIEW

**Comment by liggitt** [comment]
Lgtm, is this exercised by standard e2e runs?

**Comment by crassirostris** [comment]
> Lgtm, is this exercised by standard e2e runs?

Logs ingestion -- yes, collecting metrics -- not sure

/approve no-issue

@loburm could you please take a look and lgtm?

File: cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml [hunk 1]
```diff
@@ -82,17 +92,14 @@ spec:
               fi;
       # BEGIN_PROMETHEUS_TO_SD
       - name: prometheus-to-sd-exporter
-        image: gcr.io/google-containers/prometheus-to-sd:v0.1.3
+        image: gcr.io/google-containers/prometheus-to-sd:v0.2.1
```
Comment (loburm) [inline cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml]: Unfortunately this version have some problems related to noisy logging. I have just released v0.2.2, please switch to it.

File: cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml [hunk 2]
```diff
@@ -82,17 +92,14 @@ spec:
...
         command:
           - /monitor
```
Comment (loburm) [inline cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml]: Flags component, target-port and whitelisted-metrics are deprecated. Please use --source instead: https://github.com/GoogleCloudPlatform/k8s-stackdriver/blob/master/prometheus-to-sd/main.go#L68

Comment (loburm) [inline cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml]: Now in newer version of prometheus-to-sd it's possible to specify pod name and pod namespace. Here is an example:
https://github.com/GoogleCloudPlatform/k8s-stackdriver/blob/master/prometheus-to-sd/kubernetes/prometheus-to-sd-kube-state-metrics.yaml

**Review by loburm — CHANGES_REQUESTED** [review]

File: cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml (hunk 2 above)
Comment (crassirostris) [reply cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml]: Will it still work? WDYT about leaving it as it is and following-up in a separate PR?

Comment (crassirostris) [reply cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml]: Ditto

**Comment by tallclair** [comment]
@loburm - Thank you for the review. I'd like to keep these other changes to a minimum, because I'm planning on cherrypicking this change. Would it be OK to address the version bump & new flags in a later PR? (maybe you or @crassirostris could tackle that?)

**Comment by crassirostris** [comment]
@tallclair I checked and we can even skip updating the version for now, just remove the certs, it works

**Comment by tallclair** [comment]
Okay, reverted version change, just leaving the service account & SSL cert changes to unblock the PodSecurityPolicy config.

File: cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml [hunk 3]
```diff
@@ -89,10 +99,8 @@ spec:
           - --target-port=31337
           - --stackdriver-prefix={{ prometheus_to_sd_prefix }}/addons
           - --api-override={{ prometheus_to_sd_endpoint }}
+          - --source=
```
Comment (crassirostris) [inline cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml]: Why this change?

**Comment by crassirostris** [comment]
Also, please update the release note

File: cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml (hunk 3 above)
Comment (tallclair) [reply cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml]: Oo, stray change. Thanks for catching this!
